File: El Mamoun/Lv 1/Problem 42.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/*
Write a program to calculate the task duration in seconds and print it on screen
 Given the time duration of a task in the number of days, hours, minutes, and seconds,
*/
namespace Problem_42
{
    internal class Program
    {

       struct stTaskDurationInfo
        {
            public int NumberOfDays;
            public int NumberOfHours;
            public int NumberOfMinutes;
            public int NumberOfSeconds;
        }

       static stTaskDurationInfo ReadTaskDurationInfo()
        {
            stTaskDurationInfo TaskDurationInfo;

            Console.WriteLine("Please Enter Number of Days?: ");
            TaskDurationInfo.NumberOfDays = Convert.ToInt32(Console.ReadLine());

            Console.WriteLine("Please Enter Number of Hours?: ");
            TaskDurationInfo.NumberOfHours = Convert.ToInt32(Console.ReadLine());

            Console.WriteLine("Please Enter Number of Minutes?: ");
            TaskDurationInfo.NumberOfMinutes = Convert.ToInt32(Console.ReadLine());

            Console.WriteLine("Please Enter Number of Seconds?: ");
            TaskDurationInfo.NumberOfSeconds = Convert.ToInt32(Console.ReadLine());

            return TaskDurationInfo;

        }

       static float CalculteTaskDuration(stTaskDurationInfo TaskDurationInfo)
        {

            float Result = TaskDurationInfo.NumberOfDays * 24 * 60 * 60 + TaskDurationInfo.NumberOfHours * 60 * 60 +
                TaskDurationInfo.NumberOfMinutes * 60 + TaskDurationInfo.NumberOfSeconds;

            return Result;


        }


        static void Main(string[] args)
        {

            Console.WriteLine(CalculteTaskDuration(ReadTaskDurationInfo()));         
            Console.ReadKey();
        }
    }
}

File: El Mamoun/Lv 1/Problem 42.cpp
#include<iostream>
using namespace std;

/*
Write a program to calculate the task duration in seconds and print it on screen
 ?Given the time duration of a task in the number of days, hours, minutes, and seconds,
*/

int ReadPostiveNumber(string message)
{
	int number{ 0 };

	do
	{

		cout << message;
		cin >> number;

	} while (number < 0);

	return number;
}

struct stTaskDuration
{
	float Days;
	float Hours;
	float Minutes;
	float Seconds;
};

stTaskDuration ReadTaskDuration()
{
	stTaskDuration TaskDuration;

	TaskDuration.Days    = ReadPostiveNumber("Please Enter Number of Days? ");
	TaskDuration.Hours   = ReadPostiveNumber("Please Enter Number of Hours? "); 
	TaskDuration.Minutes = ReadPostiveNumber("Please Enter Number of Miutes? ");	 
	TaskDuration.Seconds = ReadPostiveNumber("Please Enter Number of Seconds? ");

	return TaskDuration;
}

int TaskDurationInSeconds(stTaskDuration TaskDuration)
{
	int TaskDurationPerSecond;

	TaskDurationPerSecond = TaskDuration.Days * 24 * 60 * 60;
	TaskDurationPerSecond += TaskDuration.Hours * 60 * 60;
	TaskDurationPerSecond += TaskDuration.Minutes * 60;
	TaskDurationPerSecond += TaskDuration.Seconds;

	return TaskDurationPerSecond;
}


int main()
{

	cout << "Result = " << TaskDurationInSeconds(ReadTaskDuration());

	return 0;
}